build-logic/common-plugins/src/main/kotlin/org.xtclang.build.java.gradle.kts

```diff
@@ -0,0 +1,134 @@
+import XdkBuildLogic.Companion.DEFAULT_JAVA_BYTECODE_VERSION
+import org.gradle.api.tasks.testing.logging.TestLogEvent.STANDARD_OUT
+import org.gradle.api.tasks.testing.logging.TestLogEvent.STANDARD_ERROR
+import org.gradle.api.tasks.testing.logging.TestLogEvent.SKIPPED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.STARTED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.PASSED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED
+
+import java.nio.charset.StandardCharsets.UTF_8
+
+plugins {
+    id("org.xtclang.build.xdk.versioning")
+    java
+}
+
+private val pprefix = "org.xtclang.java"
+private val lintProperty = "$pprefix.lint"
+
+private val jdkVersion: Provider<Int> = provider {
+    getXdkPropertyInt("$pprefix.jdk", DEFAULT_JAVA_BYTECODE_VERSION)
+}
+
+java {
+    toolchain {
+        val xdkJavaVersion = JavaLanguageVersion.of(jdkVersion.get())
+        val buildProcessJavaVersion = JavaLanguageVersion.of(JavaVersion.current().majorVersion.toInt())
+        if (!buildProcessJavaVersion.canCompileOrRun(xdkJavaVersion)) {
+            throw buildException("Error in Java Toolchain config. The builder can't compile requested Java version: $xdkJavaVersion")
+        }
+        logger.info("$prefix Java Toolchain config; binary format version: 'JDK $xdkJavaVersion' (build process version: 'JDK $buildProcessJavaVersion')")
+        languageVersion.set(xdkJavaVersion)
+    }
+}
+
+testing {
+    suites {
+        @Suppress("UnstableApiUsage") val test by getting(JvmTestSuite::class) {
+            useJUnitJupiter()
+        }
+    }
+}
+
+tasks.withType<JavaExec>().configureEach {
+    inputs.property("jdkVersion", jdkVersion);
+    logger.info("$prefix Configuring JavaExec task $name from toolchain (Java version: ${java.toolchain.languageVersion})")
+    javaLauncher.set(javaToolchains.launcherFor(java.toolchain))
+    if (enablePreview()) {
+        jvmArgs("--enable-preview")
+    }
+    doLast {
+        logger.info("$prefix JVM arguments: $jvmArgs")
+    }
+}
+
+//val assemble by tasks.existing
+val checkWarnings by tasks.registering {
+    if (!getXdkPropertyBoolean(lintProperty, false)) {
+        val lintPropertyHasValue = isXdkPropertySet(lintProperty)
+        if (lintPropertyHasValue) {
```

Comment:
What is the point of these warnings? How should I (or Dima) act on that? If you believe some compiler option needs to be on, let's turn it on

Reply:
The point is that we should fix the millions of Java warnings in our Java code base, for example use of Raw types. This is the reminder. The alternative is to spew them out on the screen, and die on error. All other alternatives (silently ignore Java code with lots of warnings) have gotten me, as someone who wants to work with the code base in trouble, repeatedly.  

If you don't want to see the message, use of the semi persistent ways to set the lint property of false in your environment, as described in the documentation.  Add an export of its env variable equivalent to your shell .rc profile or something. This is a default setting that is warranted for several reasons, and we should strive to write warning free Java code.

Reply:
I haven't had a single issues caused by ignoring any of those warnings, and Cam and I have been working on this for six years now. My vote is to preserve a status quo and let the build behave as it does today in this regard. I would be happy if we had a flag to turn those warnings on - may be at some point we can hire an internal to go and clean those up :)